src/meta-srv/src/election.rs

```diff
@@ -14,12 +14,21 @@
 
 pub mod etcd;
 
+use etcd_client::LeaderKey;
+use tokio::sync::broadcast::Receiver;
+
 use crate::error::Result;
 
 pub const LEASE_SECS: i64 = 3;
 pub const KEEP_ALIVE_PERIOD_SECS: u64 = LEASE_SECS as u64 * 2 / 3;
 pub const ELECTION_KEY: &str = "__meta_srv_election";
 
+#[derive(Clone)]
+pub enum LeaderChangeMessage {
+    Elected(Box<LeaderKey>),
+    StepDown(Box<LeaderKey>),
```

Comment:
Why do we need to box the LeaderKey?

Reply:
Because broadcast channel needs cloneable elements. Though LeaderKey is cloneable, we only need to read it. So I box it here to get rid of the actual object clone.

Reply:
Clone of `Box<T>` allocates a new instance of `Box<T>`. Maybe you should use Arc?